### PR TITLE
Docker activity health check

### PR DESCRIPTION
**Pull request by NadeeshDilanga**
This has the health checker implementation for docker activity plugin.

### CODE REVIEW

File: taverna-docker-activity/src/main/java/org/apache/taverna/activities/docker/DockerActivityHealthChecker.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public boolean canVisit(Object o) {
+        return o instanceof DockerActivity;
+    }
+
+    @Override
+    public VisitReport visit(Object o, List<Object> list) {
+        DockerActivity activity = (DockerActivity) o;
+        DockerContainerConfiguration contCfg = activity.getContainerConfiguration();
+        String containerName = contCfg.getName();
+        boolean createValid = ValidationUtil.validateCreateContainer(contCfg, containerName);
```
Comment (alaninmcr) [inline taverna-docker-activity/src/main/java/org/apache/taverna/activities/docker/DockerActivityHealthChecker.java]: Does validateCreateContainer also destroy the container after it has been created? If not, then repeated validation could create a lot of useless containers

File: taverna-docker-activity/src/main/java/org/apache/taverna/activities/docker/DockerActivityHealthChecker.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        List<VisitReport> reports = new ArrayList<VisitReport>();
+
+        if (createValid) {
+            reports.add(new VisitReport(HealthCheck.getInstance(), activity,
+                    "Docker create container operation is healthy",
+                    HealthCheck.NO_PROBLEM, VisitReport.Status.OK));
+        } else {
+            reports.add(new VisitReport(HealthCheck.getInstance(), activity,
+                    "REST Activity - bad configuration",
```
Comment (alaninmcr) [inline taverna-docker-activity/src/main/java/org/apache/taverna/activities/docker/DockerActivityHealthChecker.java]: Why does it say REST activity? Also the message should be more meaningful.

File: taverna-docker-activity/src/main/java/org/apache/taverna/activities/docker/DockerActivityHealthChecker.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                    HealthCheck.INVALID_CONFIGURATION, VisitReport.Status.SEVERE));
+        }
+
+        // collection all reports together
+        VisitReport.Status worstStatus = VisitReport.getWorstStatus(reports);
+        VisitReport report = new VisitReport(HealthCheck.getInstance(), activity,
+                "REST Activity Report", HealthCheck.NO_PROBLEM, worstStatus, reports);
```
Comment (alaninmcr) [inline taverna-docker-activity/src/main/java/org/apache/taverna/activities/docker/DockerActivityHealthChecker.java]: Why REST activity?

Comment (NadeeshDilanga) [inline taverna-docker-activity/src/main/java/org/apache/taverna/activities/docker/DockerActivityHealthChecker.java]: That error message not meant to be there. It was my bad to to able to remove it. Will change it.

**Comment by NadeeshDilanga** [comment]
The create container validation will check whether the input data for container creation meets the necessities of docker only. (i.e there cannot be same name container already in running mode.)

It does not actually create a container. And in my personal opinion, we should not do that. Because here we are only suppose to validate. Plus, create container is a heavy task and it is mostly depends on how weight your image is. If it is too heavy, it takes lot time to start.

**Pull request merged**